fenix/app/src/test/java/org/mozilla/fenix/shopping/middleware/ProductAnalysisMapperTest.kt

```diff
@@ -113,14 +113,29 @@ class ProductAnalysisMapperTest {
     }
 
     @Test
-    fun `WHEN product id is null THEN it is mapped to no analysis present`() {
```

Comment:
do we also need tests to check AnalysisStatusDto and if the loading analysis card is present

Reply:
indeed, there are now tests in `ReviewQualityCheckStoreTest` for the analysis status. 

As for loading, I thought about it but it's not straightforward at this moment AFAIK. It's an intermediate state between the initial and the ending success/error states. It would involve observing the state in the test and asserting that there's loading in the middle. Not sure how to do that, let me know if you have any ideas. Also, depends on how pedantic we want to be about asserting the intermediate state.